.lattice/bin/checks/_range.sh

```
@@ -0,0 +1,13 @@
+# Shared: resolve a root-safe review base/range for git-based checks.
+# Sourced, not executed. Sets:
+#   LATTICE_DIFF_BASE  — diff HEAD against this (origin/main, else the empty tree)
+#   LATTICE_LOG_RANGE  — revision range for git log (origin/main..HEAD, else HEAD)
+# The empty-tree fallback keeps diffs valid on an initial commit or a depth-one
+# checkout that lacks origin/main (rather than the invalid HEAD~1..HEAD).
+if git rev-parse --verify -q origin/main >/dev/null 2>&1; then
+  LATTICE_DIFF_BASE="origin/main"
+  LATTICE_LOG_RANGE="origin/main..HEAD"
+else
+  LATTICE_DIFF_BASE="$(git hash-object -t tree /dev/null)"
```

Shellcheck (0.11.0)

[warning] LATTICE_DIFF_BASE appears unused. Verify use (or export if used externally).

(SC2034)